packages/vite/plugins/nx-tsconfig-paths.plugin.ts

```diff
@@ -31,59 +69,102 @@ There should at least be a tsconfig.base.json or tsconfig.json in the root of th
+        tsConfigPathsFallback = rootLevelParsed;
         matchTsPathFallback = createMatchPath(
           rootLevelParsed.absoluteBaseUrl,
           rootLevelParsed.paths,
           ['main', 'module']
         );
       }
     },
-    resolveId(source: string) {
+    resolveId(importPath: string) {
       let resolvedFile: string;
       try {
-        resolvedFile = matchTsPathEsm(source);
+        resolvedFile = matchTsPathEsm(importPath);
       } catch (e) {
         logIt('Using fallback path matching.');
-        resolvedFile = matchTsPathFallback?.(source);
+        resolvedFile = matchTsPathFallback?.(importPath);
       }
 
       if (!resolvedFile) {
-        logIt(`Unable to resolve ${source} with tsconfig paths`);
+        if (tsConfigPathsEsm || tsConfigPathsFallback) {
+          logIt(
+            `Unable to resolve ${importPath} with tsconfig paths. Using fallback file matching.`
+          );
+          resolvedFile =
+            loadFileFromPaths(tsConfigPathsEsm, importPath) ||
+            loadFileFromPaths(tsConfigPathsFallback, importPath);
+        } else {
+          logIt(`Unable to resolve ${importPath} with tsconfig paths`);
+        }
       }
 
+      logIt(`Resolved ${importPath} to ${resolvedFile}`);
       return resolvedFile;
     },
   };
-}
 
-function getTsConfig(preferredTsConfigPath: string): string {
-  return [
-    resolve(preferredTsConfigPath),
-    resolve(join(workspaceRoot, 'tsconfig.base.json')),
-    resolve(join(workspaceRoot, 'tsconfig.json')),
-  ].find((tsPath) => {
-    if (existsSync(tsPath)) {
-      logIt('Found tsconfig at', tsPath);
-      return tsPath;
+  function getTsConfig(preferredTsConfigPath: string): string {
+    return [
+      resolve(preferredTsConfigPath),
+      resolve(join(workspaceRoot, 'tsconfig.base.json')),
+      resolve(join(workspaceRoot, 'tsconfig.json')),
+    ].find((tsPath) => {
+      if (existsSync(tsPath)) {
+        logIt('Found tsconfig at', tsPath);
+        return tsPath;
+      }
+    });
+  }
+
+  function logIt(...msg: any[]) {
+    if (process.env.NX_VERBOSE_LOGGING === 'true' || options?.debug) {
+      console.debug('\n[Nx Vite TsPaths]', ...msg);
     }
-  });
-}
+  }
 
-function logIt(...msg: any[]) {
-  if (process.env.NX_VERBOSE_LOGGING === 'true') {
-    console.debug('[Nx Vite TsPaths]', ...msg);
+  function loadFileFromPaths(
+    tsconfig: ConfigLoaderSuccessResult,
+    importPath: string
+  ) {
+    logIt(
+      `Trying to resolve file from config in ${tsconfig.configFileAbsolutePath}`
+    );
+    let resolvedFile: string;
+    for (const alias in tsconfig.paths) {
+      const paths = tsconfig.paths[alias];
+
+      const normalizedImport = alias.replace(/\/\*$/, '');
+
+      if (importPath.startsWith(normalizedImport)) {
+        const path = (tsconfig.absoluteBaseUrl, paths[0].replace(/\/\*$/, ''));
+        resolvedFile = findFile(importPath.replace(normalizedImport, path));
+      }
+    }
+
+    return resolvedFile;
```

Comment:
Do we need a `resolvedFile || null` here since the `null` indicates to use other resolveId?

Reply:
Yeah I'll default to null when we don't find one, so the file keeps processing 👍